plugin/src/main/java/org/opensearch/ml/processor/MLInferenceSearchResponseProcessor.java

```diff
@@ -0,0 +1,678 @@
+
+        /**
+         * Constructs a new instance of the Factory class.
+         *
+         * @param client           the Client instance to be used by the Factory
+         * @param xContentRegistry the xContentRegistry instance to be used by the Factory
+         */
+        public Factory(Client client, NamedXContentRegistry xContentRegistry) {
+            this.client = client;
+            this.xContentRegistry = xContentRegistry;
+        }
+
+        /**
+         * Creates a new instance of the MLInferenceSearchResponseProcessor.
+         *
+         * @param processorFactories a map of processor factories
+         * @param processorTag       the tag of the processor
+         * @param description        the description of the processor
+         * @param ignoreFailure      a flag indicating whether to ignore failures or not
+         * @param config             the configuration map for the processor
+         * @param pipelineContext    the pipeline context
+         * @return a new instance of the MLInferenceSearchResponseProcessor
+         */
+        @Override
+        public MLInferenceSearchResponseProcessor create(
+            Map<String, Processor.Factory<SearchResponseProcessor>> processorFactories,
+            String processorTag,
+            String description,
+            boolean ignoreFailure,
+            Map<String, Object> config,
+            PipelineContext pipelineContext
+        ) {
+            String modelId = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, MODEL_ID);
+            Map<String, Object> modelConfigInput = ConfigurationUtils.readOptionalMap(TYPE, processorTag, config, MODEL_CONFIG);
+
+            List<Map<String, String>> inputMaps = ConfigurationUtils.readOptionalList(TYPE, processorTag, config, INPUT_MAP);
+            List<Map<String, String>> outputMaps = ConfigurationUtils.readOptionalList(TYPE, processorTag, config, OUTPUT_MAP);
+            int maxPredictionTask = ConfigurationUtils
+                .readIntProperty(TYPE, processorTag, config, MAX_PREDICTION_TASKS, DEFAULT_MAX_PREDICTION_TASKS);
+            boolean ignoreMissing = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, IGNORE_MISSING, false);
+            String functionName = ConfigurationUtils
+                .readStringProperty(TYPE, processorTag, config, FUNCTION_NAME, FunctionName.REMOTE.name());
+            boolean override = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, OVERRIDE, false);
+            boolean oneToOne = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, ONE_TO_ONE, false);
+
+            String modelInput = ConfigurationUtils.readOptionalStringProperty(TYPE, processorTag, config, MODEL_INPUT);
+
+            // if model input is not provided for remote models, use default value
+            if (functionName.equalsIgnoreCase("remote")) {
+                modelInput = (modelInput != null) ? modelInput : DEFAULT_MODEl_INPUT;
```

Comment:
You are mixing upper case and lower case in variable `DEFAULT_MODEl_INPUT` :) (Darn how did I find it...)

Reply:
Ah it should be 'DEFAULT_MODEL_INPUT'. Thanks!